lib/rucio/richclient.py

```diff
@@ -0,0 +1,283 @@
+        verbose: bool = False,
+        console: Optional[Console] = None
+) -> logging.Logger:
+    """
+    Factory method to set logger with handlers.
+    :param module_name: __name__ of the module that is calling this method
+    :param logger_name: name of the logger, typically name of the module.
+    :param logger_level: if not given, fetched from config.
+    :param verbose: verbose option set in bin/rucio
+    :param console: rich console object
+    :returns: logger with RichHandler
+    """
+    # helper method for cfg check
+    def _force_cfg_log_level(cfg_option: str) -> bool:
+        cfg_forced_modules = config_get('logging', cfg_option, raise_exception=False, default=None, clean_cached=True, check_config_table=False)
+        if cfg_forced_modules and module_name is not None:
+            if re.match(str(cfg_forced_modules), module_name):
+                return True
+        return False
+
+    # creating log
+    if not logger_name:
+        if not module_name:
+            logger_name = 'usr'
+        else:
+            logger_name = module_name.split('.')[-1]
+    logger = logging.getLogger(logger_name)
+
+    # extracting the log level
+    if not logger_level:
+        logger_level = logging.INFO
+        if verbose:
+            logger_level = logging.DEBUG
+
+        # overriding by the config
+        cfg_levels = (logging.DEBUG, logging.INFO, logging.WARNING, logging.ERROR)
+        for level in cfg_levels:
+            cfg_opt = 'forceloglevel' + logging.getLevelName(level)
+            if _force_cfg_log_level(cfg_opt):
+                logger_level = level
+
+    # setting the log level
+    logger.setLevel(logger_level)
+
+    # preferred logger handling
+    def add_handler(logger: logging.Logger,
+                    console: Optional[Console] = None,
+                    verbose: bool = False) -> None:
+
+        def time_formatter(timestamp: datetime) -> Text:
+            formatted_time = timestamp.strftime("%Y-%m-%d %H:%M:%S")
+            milliseconds = timestamp.microsecond // 1000
+            return Text(f"[{formatted_time},{milliseconds:03d}]")
+
+        console = console or Console()
+        highlighter = ReprHighlighter()
+        handler = RichHandler(rich_tracebacks=True, markup=True, show_path=verbose, show_time=verbose, console=console, tracebacks_width=min(console.width, MAX_TRACEBACK_WIDTH),
+                              tracebacks_word_wrap=True, log_time_format=time_formatter, highlighter=highlighter)
+        logger.addHandler(handler)
+
+    # setting handler and formatter
+    if not logger.handlers:
+        console = console or Console()
+        add_handler(logger, console, verbose)
+
+    return logger
+
+
+def generate_table(table_format: box.Box, headers: list[str], rows: Sequence[Union[tuple[Any, Any], list[Any]]],
+                   col_alignments: Optional[list[JustifyMethod]] = None, show_header: bool = True,
+                   row_styles: Optional[list[str]] = None) -> Table:
```

Comment:
1. Type hints add some noise. When the definition becomes to long and needs to be wrapped, typically it becomes preferable to have one parameter per line. Especially so when some of the hints are long and complex.

    ```python
    def generate_table(
        table_format: box.Box,
        headers: list[str],
        rows: Sequence[Union[tuple[Any, Any], list[Any]]],
        col_alignments: Optional[list[JustifyMethod]] = None,
        show_header: bool = True,
        row_styles: Optional[list[str]] = None
    ) -> Table:
    ```

 2. I think the `tabulate.tabulate()` signature makes more sense.

    1. `table_format` should become a keyword parameter with a default value of `CLITheme.TABLE_FMT`.
    2. `headers` should also become a keyword parameter with a default value of `None`. It should be combined together with `show_header`: if set, then the arguments contains the header names; otherwise, don’t show the header at all.

 3. Maybe rows should be `Sequence[Sequence[Any]]`? `headers` should definitely be `Sequence[str]`. Probably others as well. Let’s discuss in person.